modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/AccessTokenHandler.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * 
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.mdm.agent.proxy;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.SharedPreferences.Editor;
+import android.os.AsyncTask;
+import android.util.Log;
+import org.apache.commons.codec.binary.Base64;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * After receiving authorization code client application can use this class to
+ * obtain access token
+ */
+public class AccessTokenHandler extends Activity {
+	private static final String TAG = "AccessTokenHandler";
+	private String username = null;
+	private String password = null;
+	private String clientID = null;
+	private String clientSecret = null;
+	private String tokenEndPoint = null;
+
+	public AccessTokenHandler(String clientID, String clientSecret, String username,
+	                          String password, String tokenEndPoint, CallBack callBack) {
+		this.username = username;
+		this.password = password;
+		this.clientID = clientID;
+		this.clientSecret = clientSecret;
+		this.tokenEndPoint = tokenEndPoint;
+	}
+
+	public void obtainAccessToken() {
+		new NetworkCallTask().execute();
+	}
+
+	/**
+	 * AsyncTask to contact authorization server and get access token, refresh
+	 * token as a result
+	 */
+	private class NetworkCallTask extends AsyncTask<Void, Void, String> {
+		private String response = null;
+		private String responseCode = null;
+
+		public NetworkCallTask() {
+
+		}
+
+		@Override
+		protected String doInBackground(Void... arg0) {
+			Map<String, String> request_params = new HashMap<String, String>();
+			request_params.put("grant_type", "password");
```

Comment:
I'm seeing most of the string literals used for property names, etc are re-used in quite a few places. Let's introduce appropriate constants upon them to be able to make the code more maintainable.